tap_appsflyer/client.py

```diff
@@ -0,0 +1,122 @@
+from typing import Any, Dict, Mapping, Optional, Tuple
+
+import backoff
+import requests
+from requests import session
+from requests.exceptions import Timeout, ConnectionError, ChunkedEncodingError
+from singer import get_logger, metrics, utils
+
+from tap_appsflyer.exceptions import ERROR_CODE_EXCEPTION_MAPPING, appsflyerError, appsflyerBackoffError
+
+LOGGER = get_logger()
+REQUEST_TIMEOUT = 300
+
+def raise_for_error(response: requests.Response) -> None:
+    """Raises the associated response exception. Takes in a response object,
+    checks the status code, and throws the associated exception based on the
+    status code.
+
+    :param resp: requests.Response object
+    """
+    try:
+        response_json = response.json()
+    except Exception:
+        response_json = {}
+    if response.status_code != 200:
+        if response_json.get("error"):
+            message = "HTTP-error-code: {}, Error: {}".format(response.status_code, response_json.get("error"))
+        else:
+            message = "HTTP-error-code: {}, Error: {}".format(
+                response.status_code,
+                response_json.get("message", ERROR_CODE_EXCEPTION_MAPPING.get(
+                    response.status_code, {}).get("message", "Unknown Error")))
+        exc = ERROR_CODE_EXCEPTION_MAPPING.get(
+            response.status_code, {}).get("raise_exception", appsflyerError)
+        raise exc(message, response) from None
+
+class Client:
+    """
+    A Wrapper class.
+    ~~~
+    Performs:
+     - Authentication
+     - Response parsing
+     - HTTP Error handling and retry
+    """
+
+    def __init__(self, config: Mapping[str, Any]) -> None:
+        self.config = config
+        self._session = session()
+        self.base_url = "https://hq1.appsflyer.com"
+
+
+        config_request_timeout = config.get("request_timeout")
+        if config_request_timeout and float(config_request_timeout):
+            self.request_timeout = float(config_request_timeout)
+        else:
+            self.request_timeout = REQUEST_TIMEOUT
```

Comment:
Logically your code is correct, just a suggestion.
```suggestion
        self.request_timeout = float(config_request_timeout) if config_request_timeout else REQUEST_TIMEOUT
```